include/boost/asio/detail/socket_types.hpp

```diff
@@ -62,7 +62,9 @@
    || defined(__OpenBSD__) || defined(__linux__) \
    || defined(__EMSCRIPTEN__)
 #  include <poll.h>
-# elif !defined(__SYMBIAN32__)
+# elif !defined(__SYMBIAN32__) && !defined(__VXWORKS__ )
+#  include <sys/poll.h>
+# elif defined(__VXWORKS__ ) && (_WRS_VXWORKS_MAJOR > 6 )
```

Comment:
We added an option fake poll() that just wraps select(), for instances like ASIO streams where there is no select() alternative.